### ISSUE TITLE
Problem when limiting rows

### ISSUE THREAD

**Issue by pedrodiballin** [lucidworks/spark-solr#254]
I'm trying to load just a limited number of records when i'm performing a select. I have this code below:

```
SolrJavaRDD solrRDD = SolrJavaRDD.get(zkHost, collection, javaSparkSession.sparkContext());
SelectSolrRDD selectSolrRDD = solrRDD.rdd();

SolrQuery solrQuery = new SolrQuery().setQuery("*:*").setRows(2);
JavaRDD<SolrDocument> solrJavaRDD = new SolrJavaRDD(selectSolrRDD.query(solrQuery));

System.out.println("Count: " + solrJavaRDD.count());
```

I have setted the number of rows and the log shows me the query below:

```INFO SelectSolrRDD: Updated Solr query: q=*:*&rows=2&distrib=false&start=0&sort=id+asc&fq=_version_:[*+TO+1629626520052957184]```

But when i'm performing the count on RDD it returns all data in the collection, not only the first 2 rows i asked.

When i use the spark way, it works very well:

```
Map options = new HashMap<String, String>();
       options.put("zkhost", zkHost);
       options.put("collection", collection);
       options.put("query", "*:*");
       options.put("max_rows", "2");

JavaRDD<Row> rdd = javaSparkSession.read().format("solr").options(options).load().javaRDD();
```

This scenario doesn't fit for me because i need a SolrDocument and not a Row.

Is this a bug? Or i'm using on the wrong way?

**Comment by kiranchitturi** [comment]
SolrJavaRDD is not really well supported and does not have all the functionality. why don't you try this https://github.com/lucidworks/spark-solr/blob/dad2133c068c13572baf1dec92a0f3e1c8ba1e0c/src/main/scala/com/lucidworks/spark/rdd/SolrRDD.scala#L119-L118?

**Comment by pedrodiballin** [comment]
Do you have any code examples on how to use this class with Java? Because in the ReadMe section there is only examples for SolrJavaRDD. Even when i use just SelectSolrRDD some parameters like "start" and "rows" are not being respect.

**Comment by pedrodiballin** [comment]
I have another issue either, i'm sending the following parameter:

`options.put("solr.params", "start=5");`

It's printing the following query: q=(*:*)&qt=/select&sort=id+asc&start=5&collection=sample_test

But, the "start" parameter is not being respected. It always starts from data index 0, not from the index I requested in options

**Comment by kiranchitturi** [comment]
spark-solr is not designed for pagination. It is designed to stream data in parallel from multiple shards
